frontend/src/components/Footer/Footer.jsx

```diff
@@ -1,255 +1,263 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "react-router-dom";
 import style from "./footer.module.scss";
 
-export class Footer extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      fields: {},
-      errors: {},
-    };
-  }
-  handleValidation() {
-    let fields = this.state.fields;
-    let errors = {};
-    let formIsValid = true;
-
-    if (!fields["email"]) {
-      formIsValid = false;
-      errors["email"] = "* Email cannot be empty";
-    }
+export const Footer = () => {
+  //email state
+  const [email, setEmail] = useState("");
+
+  //setting email error
+  const [emailErr, setEmailErr] = useState({});
+
+  //setting up validations
+  const validation = () => {
+    const emailErr = {};
 
-    if (typeof fields["email"] !== "undefined") {
-      let lastAtPos = fields["email"].lastIndexOf("@");
-      let lastDotPos = fields["email"].lastIndexOf(".");
-
-      if (
-        !(
-          lastAtPos < lastDotPos &&
-          lastAtPos > 0 &&
-          fields["email"].indexOf("@@") === -1 &&
-          lastDotPos > 2 &&
-          fields["email"].length - lastDotPos > 2
-        )
-      ) {
-        formIsValid = false;
-        errors["email"] = "* Email is not valid";
-      }
+    let isValid = true;
+
+    //email
+    // eslint-disable-next-line
+    const emailRegex = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
Instead of using regex, let's use `joi-validators` only in order to be consistent everywhere :)

Reply:
@Kajol-Kumari , earlier in footer section email was validated by regex  in class component, which I had updated in functional component.I hadn't use joi-validator dependency earlier!
Have we use joi in this project ,so I could take ref.

Reply:
Hey @ShubhamSj07 please take the reference of resource form for using `joi-validator` and in order to be consistent we are going with it.